### PR TITLE
[15.0][IMP] account_banking_mandate_contact: Improve tests (remove mock)

### PR DESCRIPTION
**Pull request by victoralmau**
`account_banking_mandate_contact`: Improve tests
Changes done:
- Add _get_payment_method_information() compatibility from account_banking_mandate
- Remove mock from account_banking_mandate_contact tests

@Tecnativa

### CODE REVIEW

**Comment by pedrobaeza** [comment]
This still seems to be incorrect, but anyway, doing the mock sounds very weird and a hack. We shouldn't need to do any mock on these tests. Do you know the original intention of them? Can we remove it and fix tests without the mocks?

**Comment by victoralmau** [comment]
This fix is the same as the one done in the commit I mention and is a side effect that affects many modules causing all tests to fail. What other change do you propose to solve it?

**Comment by pedrobaeza** [comment]
Well,tests are red, so it's not working, but I'm just saying that a mock seems weird.

**Comment by victoralmau** [comment]
Extra changes done and mocks have been removed + update the PR description

File: account_payment_order/models/account_payment_method.py [hunk 1]
```diff
@@ -13,3 +14,14 @@ class AccountPaymentMethod(models.Model):
         "some payment methods.",
         default=False,
     )
+
+    @api.model
+    def _get_payment_method_information(self):
+        """Add extra codes only from account_payment_order tests."""
+        res = super()._get_payment_method_information()
+        if config["test_enable"] and self.env.context.get("test_account_payment_order"):
```
Comment (rousseldenis) [inline account_payment_order/models/account_payment_method.py]: @victoralmau That's a regression IMHO. You can't embed test feature in code... it's nonsense !

**Review by lmignon — CHANGES_REQUESTED** [review]
@victoralmau IMO, replacing mock by test feature into the original implementation is not an improvement it's a hack. Can you keep the mock plz.

**Comment by victoralmau** [comment]
`account_payment_order` changes removed.

**Review by pedrobaeza — APPROVED** [review]

**Review by rousseldenis — APPROVED** [review]

**Pull request merged**